core/src/main/scala/org/apache/spark/executor/TaskMetrics.scala

```diff
@@ -334,6 +342,14 @@ class ShuffleReadMetrics extends Serializable {
    * Number of blocks fetched in this shuffle by this task (remote or local)
    */
   def totalBlocksFetched = _remoteBlocksFetched + _localBlocksFetched
+
+  /**
+   * Total number of records read from the shuffle by this task
+   */
+  private var _recordsRead: Long = _
```

Comment:
to be more consistent with the presentation in the UI and the other fields here, should this be shuffleRecordsRead?